var/spack/repos/builtin/packages/hwloc/package.py

```diff
@@ -74,7 +74,7 @@ def configure_args(self):
             "--enable-cuda" if '+cuda' in spec else "--disable-cuda",
             "--enable-libxml2" if '+libxml2' in spec else "--disable-libxml2",
             "--enable-pci" if '+pci' in spec else "--disable-pci",
-            "--enable-static" if '+static' in spec else "--disable-static",
+            "--enable-shared" if '+shared' in spec else "--disable-shared",
```

Comment:
I am not sure what hwloc autotools' default is, but if is static libraries, `--enable-shared` would disable the static build. For clarification, we should set both.

Reply:
The default is shared. Here is the relevant part from the help:

```
  --enable-shared[=PKGS]  build shared libraries [default=yes]
  --enable-static[=PKGS]  build static libraries [default=no]
```

Using `--enable-shared` will build shared libs and not static.
Using `--disable-shared` will build static libs and not shared. 

I have checked the above with the latest only version of hwloc (`1.11.8`), which is currently the default within spack. 

I believe that setting two variants is a better approach (the use case being someone that wants to have both shared and static libraries in the same build). I understand on the other hand this is not very common within spack package specs. 

Let me know how if I should change something.

Reply:
> I believe that setting two variants is a better approach (the use case being someone that wants to have both shared and static libraries in the same build).

@pkoro Having 2 variants is not a good solution imo for one simple reason: what does it mean `hwloc~shared~static`? What is currently being done in most packages is to provide both libraries, if possible, for `+shared` and only the static one for `~shared`.

---

For your information: there's a discussion on this topic - inactive since a while - in #5269. Regarding the `shared` vs. `static` variants I am still of the opinion expressed [here](https://github.com/spack/spack/issues/5269#issuecomment-331561272) .

Reply:
Let's leave it like this for now!